### DISCUSSION TITLE
Dendrite horizontal scaling on multiple servers

### DISCUSSION THREAD

**Discussion by tilor** [matrix-org/dendrite#2469] [Help & Support]
Hi,
I found in your docs that Dendrite can run the polylith modules on separate machines. However I didn't find anything about scaling really large systems, e.g. if Dendrite can run multiple instances on multiple servers?
- Would it be possible to install the monolith version (which would be simpler) on multiple servers behind a load balancer?
- If so, would theses instances share the same database?
- Would they be able to push presence notifications to users that belong to the same domain but are (through the load balancer) connected to different physical machines?

Or do you have any docs/plans how to set up large distributed systems? 

Besides that the Matrix ecosystem looks very promising - thank you very much to all of the team!

**Comment by tobishima-mobilus** [comment]
I also want to know the same thing. Are the database(postgres) settings and nats jetstream settings in dendrite.yaml in the same destination, so are they correct?

**Comment by kareemlukitomo** [comment]
I found [these docker-compose.yaml files](https://github.com/matrix-org/dendrite/tree/v0.10.6/build/docker) in the older commits/tags from the Dendrite repo, which might be useful. The v0.10.6 tag seems to be the most recent one with a complete compose file and Dockerfiles, so it could serve as a good starting point for deploying Dendrite in polylith mode.

Additionally, I found the latest update to the Docker image for the polylith setup, which is v0.11.1 [here](https://hub.docker.com/layers/matrixdotorg/dendrite-polylith/v0.11.1/images/sha256-6f533840a0e64cf9c9d2133144f9b24a57b0d53e055039b2c928ef66b914b75d?context=explore), updated one year ago.

Hope this information helps anyone looking to deploy Dendrite in a similar setup! I'll be trying this out soon.

P.S. I'm still learning Kubernetes for HA and not an expert by any means. :b just curious to make my Matrix server HA
